smart_open/s3.py

```diff
@@ -170,27 +172,30 @@ def __init__(self, s3_object, content_length, version_id=None):
         self._object = s3_object
         self._content_length = content_length
         self._version_id = version_id
-        self.seek(0)
+        self._position = 0
+        self._body = None
 
     def seek(self, position):
         """Seek to the specified position (byte offset) in the S3 key.
 
         :param int position: The byte offset from the beginning of the key.
         """
-        self._position = position
-        range_string = make_range_string(self._position)
-        logger.debug('content_length: %r range_string: %r', self._content_length, range_string)
-
         #
         # Close old body explicitly.
-        # When first seek(), self._body is not exist. Catch the exception and do nothing.
+        # When first seek() after __init__(), self._body is not exist.
         #
-        try:
+        if self._body is not None:
             self._body.close()
-        except AttributeError:
-            pass
+        self._body = None
+        self._position = position
+
+    def _load_body(self):
+        """Build a continuous connection with the remote peer starts from the current postion.
+        """
+        range_string = make_range_string(self._position)
+        logger.debug('content_length: %r range_string: %r', self._content_length, range_string)
 
-        if position == self._content_length == 0 or position == self._content_length:
+        if self._position == self._content_length == 0 or self._position == self._content_length:
```

Comment:
I wonder if this can be simplified to:

```suggestion
        if self._position == self._content_length:
```

Because you have a situation where your conditional is `if A or B` and `B` is always true whenever `A` is.

What do you think?

Reply:
Yeah, because the `or` is a short circuit operation, the `B` will not be evaluated when the `A` is True. But I would think the verbose judgement is helpful for expressing the behaviour of the class, as the following comment stated.
```python
# 
# When reading, we can't seek to the first byte of an empty file.
# Similarly, we can't seek past the last byte.  Do nothing here.
```

Reply:
OK, you have a point. Let's leave it as is.